x/merkledb/trieview.go

```diff
@@ -148,20 +147,12 @@ func newTrieView(
 	parentTrie TrieView,
 	changes ViewChanges,
 ) (*trieView, error) {
-	sentinelNode, err := parentTrie.getEditableNode(Key{}, false /* hasValue */)
-	if err != nil {
-		if errors.Is(err, database.ErrNotFound) {
-			return nil, ErrNoValidRoot
-		}
-		return nil, err
-	}
-
 	newView := &trieView{
-		sentinelNode: sentinelNode,
-		db:           db,
-		parentTrie:   parentTrie,
-		changes:      newChangeSummary(len(changes.BatchOps) + len(changes.MapOps)),
-		tokenSize:    db.tokenSize,
+		root:       maybe.Bind(parentTrie.getRoot(), (*node).clone),
```

Comment:
this no longer calls db.lock.RLock() before grabbing the root, not sure if it could be an issue since it still rlocks the commit lock, but requires more review.